ios/browser/ui/webui/brave_wallet/wallet_page_ui.mm

```diff
@@ -29,12 +29,29 @@
 #include "brave/ios/browser/brave_wallet/brave_wallet_service_factory.h"
 #include "brave/ios/browser/brave_wallet/meld_integration_service_factory.h"
 #include "brave/ios/browser/brave_wallet/swap_service_factory.h"
+#include "brave/ios/browser/ui/webui/sanitized_image_source.h"
 #include "brave/ios/web/webui/brave_web_ui_ios_data_source.h"
 #include "brave/ios/web/webui/brave_webui_utils.h"
 #include "components/grit/brave_components_resources.h"
 #include "ios/chrome/browser/shared/model/profile/profile_ios.h"
 #include "ios/web/public/web_state.h"
 
+namespace {
+// Content-Security-Policy: frame-src chrome-untrusted://trezor-bridge/
+// chrome-untrusted://ledger-bridge/ chrome-untrusted://nft-display/
+// chrome-untrusted://line-chart-display/ chrome-untrusted://market-display/;
+// fails to work unless all the trailing slashes from each policy.
+// For some reason, on Desktop, the trailing slashes don't seem to matter???
+// So for iOS, this function will strip the trailing slashes so the CSPs apply
+// correctly
+std::string StripTrailingSlash(std::string_view url) {
+  if (!url.empty() && url.back() == '/') {
+    return std::string(url.substr(0, url.size() - 1));
+  }
+  return std::string(url);
+}
+}  // namespace
```

Comment:
`base::TrimString` is probably a better option but I assume this was copied from desktop's code?

Reply:
it is actually not copied from desktop. Desktop is somehow fine with trailing slashes from the untrusted scheme url.
have updated with base:TrimString in [a9ba31a](https://github.com/brave/brave-core/pull/32032/commits/a9ba31a0bd5df169c1b90e703785733161a1ef0c)